activerecord/lib/active_record/connection_adapters/postgresql/oid/range.rb

```diff
@@ -69,13 +69,24 @@ def type_cast_single_for_database(value)
             def extract_bounds(value)
               from, to = value[1..-2].split(",")
               {
-                from:          (value[1] == "," || from == "-infinity") ? infinity(negative: true) : from,
-                to:            (value[-2] == "," || to == "infinity") ? infinity : to,
+                from:          (value[1] == "," || from == "-infinity") ? infinity(negative: true) : unquote(from),
+                to:            (value[-2] == "," || to == "infinity") ? infinity : unquote(to),
                 exclude_start: (value[0] == "("),
                 exclude_end:   (value[-1] == ")")
               }
             end
 
+            def unquote(value)
+              if value[0] == '"' && value[-1] == '"'
+                value[1..-2].tap do |quoted_value|
+                  quoted_value.gsub!('""','"')
```

Comment:
While PostgreSQL is permissive about accepting multiple escape styles for double quotes in input (`\"` or `""`), the output format always uses `""` (see manual 8.16.6)

Reply:
Can you note the manual link on the code why we unquote two characters?